packages/medusa/src/loaders/plugins.ts

```diff
@@ -544,6 +550,28 @@ function registerModels(
   })
 }
 
+/**
+ * Runs all setup functions in a plugin. Setup functions are run before anything from the plugin is
+ * registered to the container. This is useful for running custom build logic, fetching remote
+ * configurations, etc.
+ * @param pluginDetails The plugin details including plugin options, version, id, resolved path, etc.
+ */
+async function runSetupFunctions(pluginDetails: PluginDetails): Promise<void> {
+  const files = glob.sync(`${pluginDetails.resolve}/setup/*.js`, {})
```

Comment:
suggestion: should this be picking up `ts` files as well?

Reply:
We are proccesing the built files so they will always be `*.js`